### ISSUE TITLE
cannot open backup.sh: No such file

### ISSUE THREAD

**Issue by Jdavid77** [ariaieboy/pg-backup-s3#2]
Greetings!!

Tried to use your docker image to backup my postgres server but at the moment i'm getting this error:

![Screenshot 2024-10-07 at 21 56 08](https://github.com/user-attachments/assets/cb3f9aeb-003a-4f7b-b9f2-4272057b2cc2)

Here is my CronJob configuration:

```yaml
apiVersion: batch/v1
kind: CronJob
metadata:
  name: postgresql-backup-cron-job
spec:
  schedule: "0 0 * * 0"
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: postgresql-backup-job
            image: ariaieboy/pg-backup-s3:latest@sha256:6292f7c5d95f36285306c3bbc8faaed3d95f03d14229e253f7dd62723a480ca9
            env:
              - name: S3_REGION
                value: "weur"
              - name: S3_BUCKET
                value: "postgres"
              - name: S3_PREFIX
                value: "backup"
              - name: S3_ENDPOINT
                value: "http://minio.storage.svc.cluster.local:9000"
              - name: POSTGRES_BACKUP_ALL
                value: "true"
              - name: POSTGRES_HOST
                value: "${NFS_SERVER}"
              - name: POSTGRES_EXTRA_OPTS
                value: "--schema=public --blobs"
            envFrom:
              - secretRef:
                  name: postgres-backup-secret
            imagePullPolicy: IfNotPresent
          restartPolicy: OnFailure
      backoffLimit: 3
```

Probably some misconfiguration from my part? ( the secret contains the credentials for s3 and postgres server )

**Comment by ariaieboy** [comment]
can u test `sha256:a6617c61febfba588ed52310e08025ce8c9fe5c7d4b41d10559bbea6a6a4a0ed`

**Comment by Jdavid77** [comment]
It worked!!

Appreciate the effort mate , but i think the current script has some issues. While trying to backup all databases i got an error on line 43

```bash
if [ "${S3_ENDPOINT}" == "**None**" ]; then
```

and line 63

```bash
if [ "${POSTGRES_BACKUP_ALL}" == "true" ]; then
```

where the operator should be `=`and not `==`.



Was also getting an error on:

```bash
if [ "${POSTGRES_DATABASE}" = "**None**" -a "${POSTGRES_BACKUP_ALL}" != "true" ]; then
```

which I replaced it with:

```bash
if [ "${POSTGRES_DATABASE}" = "**None**" ] && [ "${POSTGRES_BACKUP_ALL}" != "true" ]; then
```


Made a fork of your current repo where I fixed these issues and did some Dockerfile optimisation , mainly using alpine base image. Hope you don't mind it ;)

[Repo](https://github.com/Jdavid77/pg-backup-s3)

Have a nice one!
